src/main/java/seedu/address/ui/ViewWindow.java

```diff
@@ -0,0 +1,166 @@
+package seedu.address.ui;
+
+import java.util.Comparator;
+import java.util.logging.Logger;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.FlowPane;
+import javafx.stage.Stage;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a controller for a page to view all information about a contact.
+ */
+public class ViewWindow extends UiPart<Stage> {
+
+    private static final String FXML = "ViewWindow.fxml";
+
+    private static final Logger logger = LogsCenter.getLogger(ViewWindow.class);
+
+    private static final int FIRST_PERSON_INDEX = 0;
+
+    @FXML
+    private Label name;
+    @FXML
+    private Label phone;
+    @FXML
+    private Label address;
+    @FXML
+    private Label email;
+    @FXML
+    private FlowPane tags;
+    @FXML
+    private Label dateOfLastVisit;
+    @FXML
+    private Label emergencyContact;
+    @FXML
+    private Label remark;
+
+    /**
+     * Creates a new ViewWindow.
+     * Root of the ViewWindow will always be a new Stage.
+     */
+    public ViewWindow() {
+        super(FXML, new Stage());
+    }
+
+    /**
+     * Updates the ViewWindow with all the necessary information about a person.
+     *
+     * @param personList The filtered {@code ObservableList} from {@code Logic}. Will only view 1 {@code Person}.
+     */
+    public void view(ObservableList<Person> personList) {
```

Comment:
Just wondering. Why does this method accept a list instead of just accepting the person straight up. Isn't just complicating things?

Reply:
This method is called in the `MainWindow` class, which is at quite a high level of abstraction. Within the UI elements, there are generally no mentions or references to any `Logic` objects like the `Person` class, and so this method is called as `view(logic.getFilteredPersonList())`. If I were to make this accept a `Person`, the method call would have to invoke the `.get()` within `MainWindow`, which I think makes the SLAP worse.